src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+
+import { Person } from '../../types/Person';
+import { PersonLink } from './PersonLink';
+import { getVisiblePeople } from '../../utils/getVisiblePeople';
+import { SortLink } from '../SortLink';
+
+type Props = {
+  selectedPersonSlug: string;
+  people: Person[],
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  selectedPersonSlug,
+  people,
+}) => {
+  const [searchParams] = useSearchParams();
+  const centuries = searchParams.getAll('centuries');
+  const sexParam = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const sort = searchParams.get('sort') as keyof Person;
+  const isReversed = searchParams.get('order') === 'desc';
+
+  const visiblePeople = useMemo(() => (
+    getVisiblePeople(people, centuries, sexParam, query, sort, isReversed)
+  ), [people, centuries, sexParam, query, sort, isReversed]);
+
+  const isPeopleTableEmpty = Boolean(!visiblePeople.length);
+
+  if (isPeopleTableEmpty) {
+    return <p>There are no people matching the current search criteria</p>;
+  }
+
+  return (
+    <table
+      data-cy="peopleTable"
+      className="table is-striped is-hoverable is-narrow is-fullwidth"
+    >
+      <thead>
+        <tr>
+          <th>
+            <span className="is-flex is-flex-wrap-nowrap">
+              Name
+              <SortLink sort="name" />
+            </span>
+          </th>
+
+          <th>
+            <span className="is-flex is-flex-wrap-nowrap">
+              Sex
+              <SortLink sort="sex" />
+            </span>
+          </th>
+
+          <th>
+            <span className="is-flex is-flex-wrap-nowrap">
+              Born
+              <SortLink sort="born" />
+            </span>
+          </th>
+
+          <th>
+            <span className="is-flex is-flex-wrap-nowrap">
+              Died
+              <SortLink sort="died" />
+            </span>
+          </th>
+
+          <th>Mother</th>
+          <th>Father</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {visiblePeople.map(person => {
+          const {
+            sex, born, died, motherName, fatherName, slug,
+          } = person;
+
+          const isSelected = slug === selectedPersonSlug;
+
+          const mother = people.find(parent => parent.name === motherName);
+          const father = people.find(parent => parent.name === fatherName);
```

Comment:
this code fires on each rerender, preparing the full array right after fetching the data would be better for optimization